src/components/Layout.tsx

```diff
@@ -7,14 +7,17 @@ import {
   Notifications,
   Panel,
 } from "@/components";
-import { useHandleQueryInUrl, usePageContext } from "@/hooks";
+import { usePageContext } from "@/hooks";
+import { useQueryInSearchParams } from "@/hooks/useQueryInSearchParams";
 import type { ReactNode } from "react";
 import { LegacyDappLinkBanner } from "./LegacyDappLinkBanner";
 import StyledComponentsRegistry from "./StyledComponentsRegistry";
+import { useFiltersInSearchParams } from "@/hooks/useFiltersInSearchParams";
 
 export function Layout({ children }: { children: ReactNode }) {
   const { page } = usePageContext();
-  useHandleQueryInUrl();
+  useQueryInSearchParams();
```

Comment:
this layout is only rendered once and is persisted across page changes. so it makes sense to do the hook invocations here